de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorStatusChangeHandler.java

```diff
@@ -206,4 +264,106 @@ public void setEnabled(boolean enabled) {
       this.enabled = true;
     }
   }
+
+  /**
+   * Queues a viewport adjustment for the given path using the given range and selection as
+   * parameters for the viewport adjustment. If an editor is given, it will be used for the viewport
+   * adjustment.
+   *
+   * @param path the path of the editor
+   * @param editor the editor to queue a viewport adjustment for
+   * @param range the line range used for the viewport adjustment
+   * @param selection the text selection used for the viewport adjustment
+   */
+  void queueViewPortChange(
+      @NotNull String path,
+      @Nullable Editor editor,
+      @Nullable LineRange range,
+      @Nullable TextSelection selection) {
+
+    QueuedViewPortChange requestedViewportChange =
+        new QueuedViewPortChange(editor, range, selection);
+
+    QueuedViewPortChange previouslyQueuedRequest =
+        queuedViewPortChanges.put(path, requestedViewportChange);
+
+    if (previouslyQueuedRequest != null) {
+      if (log.isTraceEnabled()) {
+        log.trace(
+            "Dropping previously queued viewport change request for "
```

Comment:
I am not sure whether this log is really necessary. Especially because it introduces a log of boilerplate code with the `QueuedViewPortChange.toString()` method. Any opinions?